src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -173,86 +211,192 @@ class HRVOTest : public ::testing::Test
+        Point dest(7.0, static_cast<double>(i) * 0.25);
+        friendly_start_dest_points.emplace_back(std::make_pair(position, dest));
     }
+    instantiate_robots_in_world(friendly_start_dest_points, {});
 }
 
-TEST_F(HRVOTest, 1_robot_in_line)
+TEST_F(HRVOTest, div_a_friendly_and_enemy_robot_performance_test)
 {
-    simulator.addAgent(Vector2(-4.f, 0.f), simulator.addGoal(Vector2(4.f, 0.f)));
-}
+    // Add robots around circle with one in the center
+    const int num_robots_per_team = 11;
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points;
+    std::vector<std::pair<Point, Vector>> enemy_position_velocity_pairs;
+    Vector robot_offset(0.0, -0.4);
+    Vector team_offset(8.0, 0.0);
+    for (std::size_t i = 0; i < num_robots_per_team; ++i)
+    {
+        Point friendly_pos = Point() + 2 * i * robot_offset;
+        friendly_start_dest_points.emplace_back(
+            std::make_pair(friendly_pos, friendly_pos + team_offset));
 
-TEST_F(HRVOTest, 1_robot_two_goals)
-{
-    simulator.addAgent(Vector2(-4.f, 0.f), simulator.addGoalPositions(
-                                               {Vector2(4.f, 0.f), Vector2(-4.f, 0.f)}));
+        Point enemy_pos = Point() + (2 * i + 1) * robot_offset + team_offset;
+        enemy_position_velocity_pairs.emplace_back(
+            std::make_pair(enemy_pos, Vector(-1.0, 0.0)));
+    }
+    instantiate_robots_in_world(friendly_start_dest_points,
+                                enemy_position_velocity_pairs);
 }
 
-TEST_F(HRVOTest, 1_robot_moving_in_square)
+TEST_F(HRVOTest, div_a_friendly_and_enemy_robot_performance_test_moving_across)
 {
-    simulator.addAgent(
-        Vector2(-4.f, -4.f),
-        simulator.addGoalPositions({Vector2(4.f, -4.f), Vector2(4.f, 4.f),
-                                    Vector2(-4.f, 4.f), Vector2(-4.f, -4.f)}));
+    // Add robots around circle with one in the center
+    const int num_robots_per_team = 11;
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points;
+    std::vector<std::pair<Point, Vector>> enemy_position_velocity_pairs;
+    Vector enemy_robot_offset(0.0, -1.0);
+    Vector friendly_robot_offset(0.4, 0.0);
+    Vector dest_offset(0.0, -10.5);
+    for (std::size_t i = 0; i < num_robots_per_team; ++i)
+    {
+        Point friendly_pos = Point(0.0, 0.0) + i * friendly_robot_offset;
+        friendly_start_dest_points.emplace_back(
+            std::make_pair(friendly_pos, friendly_pos + dest_offset));
+
+        Point enemy_pos = Point(-1.0, 0.0) + i * enemy_robot_offset;
+        enemy_position_velocity_pairs.emplace_back(
+            std::make_pair(enemy_pos, Vector(1.5, 0.0)));
+    }
+    instantiate_robots_in_world(friendly_start_dest_points,
+                                enemy_position_velocity_pairs);
+    setSimulationTimeout(7.f);
 }
 
-TEST_F(HRVOTest, 1_robot_moving_in_square_with_final_speed)
+TEST_F(HRVOTest, friendly_robot_going_around_ball_obstacle)
```

Comment:
why is the ball the same size as the robot?

Reply:
I ended up setting all Agent radii to be the same size as the max robot radius, since the actual Agent radius is generally scaled (eg. doubled for enemy agents, or 0.5m for ball). So the visualization would've been what HRVO sees, instead of the actual robot/ball size. To make it more realistic, I made a generalization that in the visualization, all Agents are the same size as the max robot radius.